src/ui/render2d.cpp

```diff
@@ -0,0 +1,224 @@
+/**
+* Text rendering adapted from: https://learnopengl.com/#!In-Practice/Text-Rendering
+*/
+
+#include "render2d.h"
+
+#include <iostream>
+#include <glm/gtc/matrix_transform.hpp>
+#include <glm/gtc/type_ptr.hpp>
+
+#include "events.h"
+#include "shaders/shader_manager.h"
+
+Render2D::Render2D() {
+    // Setup callbacks for window size.
+    events::window_buffer_resize_event.connect([&](events::WindowSizeData d) {
```

Comment:
I made the window to fire off a resize event after all subsystems are initialized.